src/products/api/viewsets.py

```diff
@@ -1,11 +1,62 @@
-from app.permissions import AllowAny
-from orders.api.base import PurchaseViewSet
+from typing import Any, TYPE_CHECKING
+
+from drf_spectacular.utils import extend_schema
+from rest_framework.generics import get_object_or_404
+from rest_framework.mixins import RetrieveModelMixin
+from rest_framework.permissions import AllowAny
+from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet
+
+from django.http import HttpResponseRedirect
+
+from banking.selector import get_bank
+from orders.api.base import PromocodeView
 from products.api import serializers
+from products.api.serializers import PurchaseSerializer
 from products.models import Course
+from products.services.purchase_creator import PurchaseCreator
+
+if TYPE_CHECKING:
+    from rest_framework.request import Request
 
+    from orders.models import Order
 
-class CourseViewSet(PurchaseViewSet):
+
+class CourseViewSet(GenericViewSet, RetrieveModelMixin):
     lookup_field = "slug"
     serializer_class = serializers.CourseSerializer
     queryset = Course.objects.all()
     permission_classes = [AllowAny]
+
+
+class CoursePromocodeView(PromocodeView):
+    product_model = Course
+
+
+class CoursePurchaseView(APIView):
```

Comment:
```suggestion
class PurchaseView(APIView):
```

Думаю слово Course теперь лишнее в названии